code/game/machinery/doors/firedoor.dm

```diff
@@ -532,6 +578,10 @@
 			new /obj/item/electronics/firelock (targetloc)
 	qdel(src)
 
+/obj/machinery/door/firedoor/Moved()
+	. = ..()
+	check_adjacent_turfs()
```

Comment:
you will need to unregister the old turfs before this

Reply:
how would I uh do this if I dont have references to the list of turfs Ive registered

Reply:
ah I see